src/qobj/arithmetic_and_attributes.jl

```diff
@@ -514,18 +514,38 @@ Quantum Object:   type=Operator   dims=[2]   size=(2, 2)   ishermitian=true
 ```
 """
 function ptrace(QO::QuantumObject{<:AbstractArray,KetQuantumObject}, sel::Union{AbstractVector{Int},Tuple})
-    length(QO.dims) == 1 && return QO
+    ns = length(sel)
+    if ns == 0
+        return tr(QO * QO')
+    else
+        nd = length(QO.dims)
+        !all(nd .>= sel .> 0) && throw(
```

Comment:
Just to avoid extra allocations, we could use `any(>(nd), sel)`.

Reply:
okay, I changed `!all(nd .>= sel .> 0)` to `(any(>(nd), sel) || any(<(0), sel))`